### DISCUSSION TITLE
Not able to use latest version

### DISCUSSION THREAD

**Discussion by Nithac** [dolanmiu/docx#1829] [Q&A] [answered]
I was importing docx js into Salesforce org and using it in Lighting Web Component to generate a word document, I was able to use older version(5.0.2) but with new versions 7.7.0 and/or 6.0.3, I am not able to import, does this product need a license?

With version 5.0.2, I am not able to use ImageRun and also not able to use SectionType property.

**Comment by dolanmiu** [comment, marked as answer]
No licence is needed

Can you show me code to reproduce error?

**Reply by Nithac** [reply]
Here is what I am doing--
1. 
Download JS file from https://unpkg.com/docx@5.0.2/build/index.js - this version is working fine, but with this version I am not able to use ImageRun.

Here is my code - this is working fine
import { LightningElement, api,wire,track } from 'lwc';
import {loadScript} from "lightning/platformResourceLoader";
import docxImport from "@salesforce/resourceUrl/docx";

export default class TestDocument extends LightningElement {

    @api recordId;
    @track records;   

    connectedCallback(){
        Promise.all([loadScript(this, docxImport)]).then(() =>{
            this.renderButtons();
        });
    }

    renderButtons(){
        this.template.querySelector(".hidden").classList.add("not_hidden");
        this.template.querySelector(".hidden").classList.remove("hidden");
    }

    startDocumentGeneration(){  
            this.buildDocument(); 
    }

    buildDocument(){
        console.log('before document');
        let document = new docx.Document();
        console.log('after document');
        this.generateDownloadLink(document);
    }

    generateDownloadLink(documentPassed){
        docx.Packer.toBase64String(documentPassed).then(textBlob =>{
            this.downloadURL = 'data:application/vnd.openxmlformats-officedocument.wordprocessingml.document;base64,' + textBlob;
            this.template.querySelector(".slds-hide").classList.remove("slds-hide");
        });
    }
}

2.I have changed my JS download to - https://unpkg.com/docx@6.0.3/build/index.js - This version is not working
same code as above, it is failing when I try to open a document - I am getting an error " [Cannot read properties of undefined (reading 'creator')]"

3.I have changed my JS download to - https://unpkg.com/docx@6.0.3/build/index.js - This version is not working
same code as above, it is failing to **loadScript** - I am getting an error - Uncaught (in promise) undefined

**Reply by dolanmiu** [reply]
Oh yes

`docxImport ` is deprecated now, it was a feature written by another person, not me. Then they stopped maintaining that feature

It will be discarded in v8.

If you require templates, try this: https://github.com/open-xml-templating/docxtemplater#readme

**Reply by Nithac** [reply]
Thank you, I will check it.

**Reply by dolanmiu** [reply]
`docx` now has this feature out of the box! As of `v8.0.0`:

https://github.com/dolanmiu/docx/blob/master/demo/85-template-document.ts

This library can now read and write docx files.
